File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int pt = 0;
        for(int i = 0; i < s.length(); i++){
            while(true){
                if(t.length() <= pt) return false;
                if(s[i] == t[pt]){
                    pt++;
                    break;
                }
                pt++;
            }
        }
        return true;
        //  ========================
        //  https://leetcode.com/problems/is-subsequence/solutions/4075908/daily-leetcode-challenge-september-day-22-c-c-python3-kotlin
        //  원리 자체는 비슷하지만 이게 더 간단하게 작성한 코드
        int length = s.size();
        int c = 0;
        if(length == 0){
            return true;
        }
        for(int i = 0; i < t.size(); i++){
            if(s[c] == t[i]){
                c++;
            }
        }

        if(c == length){
            return true;
        }
            
        return false;
    }
};

File: 0392-is-subsequence/0392-is-subsequence.cs
public class Solution {
    public bool IsSubsequence(string s, string t) {
        //  https://leetcode.com/problems/is-subsequence/solutions/4074367/93-76-two-pointers-dp
        //  c++에서 본 풀이랑 사실상 같음
        int i = 0, j = 0;
        while(i < s.Length && j < t.Length){
            if(s[i] == t[j]){
                i++;
            }
            j++;
        }
        return i == s.Length;
    }
}